test/src/lib/update-lock.spec.ts

```diff
@@ -108,40 +112,101 @@ describe('lib/update-lock', () => {
 		});
 	});
 
+	describe('cleanupLocks', () => {
+		it('should cleanup all leftover Supervisor-created lockfiles', async () => {
+			const BASE_PATH = '/mnt/root/tmp/balena-supervisor/services';
+
+			// Setup fake filesystem & stubs
+			mockFs({
+				[BASE_PATH]: {
+					'1234567': {
+						one: {
+							'updates.lock': mockFs.file({ uid: lockfile.NOBODY_UID }),
+						},
+						two: {
+							'updates.lock': mockFs.file({ uid: 0 }),
+						},
+					},
+					'7654321': {
+						three: {
+							'updates.lock': mockFs.file({ uid: lockfile.NOBODY_UID }),
+						},
+						four: {
+							'updates.lock': mockFs.file({ uid: 0 }),
+						},
+					},
+				},
+			});
+			stub(fsUtils, 'exec').resolves({
+				stdout: `${BASE_PATH}/1234567/one/updates.lock\n${BASE_PATH}/7654321/three/updates.lock\n`,
+				stderr: '',
+			});
+
+			await updateLock.cleanupLocks();
+
+			// These directories only contain lockfiles with `nobody` UID and should be removed
+			expect(
+				await fs.readdir(path.join(BASE_PATH, '1234567', 'one')),
+			).to.have.length(0);
+			expect(
+				await fs.readdir(path.join(BASE_PATH, '7654321', 'three')),
+			).to.have.length(0);
+
+			// These directories contain lockfiles with `root` UIDs and should remain
+			expect(
+				await fs.readdir(path.join(BASE_PATH, '1234567', 'two')),
+			).to.have.length(1);
+			expect(
+				await fs.readdir(path.join(BASE_PATH, '7654321', 'four')),
+			).to.have.length(1);
+
+			mockFs.restore();
+			(fsUtils.exec as SinonStub).restore();
+		});
+	});
+
 	describe('Lock/dispose functionality', () => {
-		const lockFile = updateLock.__get__('lockFile');
 		const locksTaken = updateLock.__get__('locksTaken');
 		const dispose = updateLock.__get__('dispose');
-		const lockExistsErrHandler = updateLock.__get__('lockExistsErrHandler');
+		const lockfileErrHandler = updateLock.__get__('lockfileErrHandler');
```

Comment:
Another argument for moving the error handing to the `lockfile` module is that we can get away from using `rewire` here.

Reply:
`dispose` is not used so we can remove it, and the rest should be removable if we move the lock tracking and error management functionality to `lockfile` where I think it makes more sense.

Reply:
We still do need the `lockfileErrHandler` (that I pasted above) though, since the error message if a lockfile exists is specific to the Supervisor in that it references serviceName and appId, and requires calling `dispose` which is also an `update-lock` module method.

Reply:
Removed lockfileErrHandler and explicit tests for the `dispose` method, since it's an implementation detail.